src/config.rs

```diff
@@ -28,23 +30,16 @@ impl Configuration {
             block_build_step_ms: env::var("BLOCK_BUILD_STEP_MS")
                 .ok()
                 .or_else(|| config_map.remove("BLOCK_BUILD_STEP_MS")),
+            mode: env::var("MODE").ok().or_else(|| config_map.remove("MODE")),
```

Comment:
Maybe consider declaring string values as constants, not to use raw values in code.